book-1-orientation/chapters/ADONET_INTRO.md

```diff
@@ -9,624 +9,305 @@ This will be an interactive lesson. Your instructor will lead you through follow
+                   int deptNameColumnPosition = reader.GetOrdinal  ("DeptName");
+                   string deptNameValue = reader.GetString (deptNameColumnPosition);
+
+                   // Now let's create a new department object using the   data from the database.
+                   Department department = new Department
+                   {
+                       Id = idValue,
+                       DeptName = deptNameValue
+                   };
+
+                   // ...and add that department object to our list.
+                   departments.Add(department);
+               }
+
+               // We should Close() the reader. Unfortunately, a "using"   block won't work here.
+               reader.Close();
+
+               // Return the list of departments who whomever called this  method.
+               return departments;
+           }
+       }
+   }
+   ```
+
+   To test this, let's call this method from the `Program.Main` method. Create a new instance of a `DepartmentRepository` and call `GetAllDepartments`. Run the application.
+
+   ```csharp
+   static void Main(string[] args)
+   {
+       var departmentRepo = new DepartmentRepository();
+
+       Console.WriteLine("Getting All Departments:");
+       Console.WriteLine();
+
+       var allDepartments = departmentRepo.GetAllDepartments();
+
+       foreach (var dept in allDepartments)
+       {
+           Console.WriteLine($"{dept.Id} {dept.DeptName}");
+       }
+   }
+   ```
+
+1. Now create another method in `DepartmentRepository` that will get a single department by its Id. The method should accept an `int id` as a paramter and return a single `Department` object.
+
+   ```csharp
+   /// <summary>
+   ///  Returns a single department with the given id.
+   /// </summary>
+   public Department GetDepartmentById(int id)
+   {
+       using (SqlConnection conn = Connection)
+       {
+           conn.Open();
+           using (SqlCommand cmd = conn.CreateCommand())
+           {
+               cmd.CommandText = "SELECT DeptName FROM Department WHERE Id     = @id";
+               cmd.Parameters.Add(new SqlParameter("@id", id));
+               SqlDataReader reader = cmd.ExecuteReader();
+
+               Department department = null;
+
+               // If we only expect a single row back from the database, we    don't need a while loop.
+               if (reader.Read())
+               {
+                   department = new Department
+                   {
+                       Id = id,
+                       DeptName = reader.GetString(reader.GetOrdinal   ("DeptName"))
+                   };
+               }
+
+               reader.Close();
+
+               return department;
+           }
+       }
+   }
+   ```
+
+1. Update `Program.Main` to also call this new method
+
+   ```csharp
+   Console.WriteLine("----------------------------");
+   Console.WriteLine("Getting Department with Id 1");
+
+   var singleDepartment = departmentRepo.GetDepartmentById(1);
```

Comment:
May be a bit early to use `var`

Reply:
@askingalot I'm removing `var` and adding also your suggestion of returning an object from `AddDepartment` but I want you to know that this is a now a line of this code.....
`Department department = departmentRepo.AddDepartment(new Department { DepartmentName = "Legal" })`